### PR TITLE
.Net support and limitations documentation

### PR DESCRIPTION
**Pull request by vtikoo**
Signed-off-by: Vikas Tikoo <vikasamar@gmail.com>

### CODE REVIEW

**Comment by vtikoo** [comment]
@jxyang do we have a recommendation for `DOTNET_SYSTEM_GLOBALIZATION_INVARIANT`?

File: doc/dotnet-support.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
+# Mystikos Support for .Net
+
+### Supported .Net and ASP.Net version
+We recommend using .net 3.1 and .net 5.
```
Comment (jxyang) [inline doc/dotnet-support.md:4]: What about the asp.net version support?

File: doc/dotnet-support.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+myst-lldb, a lldb extension is distributed with Mystikos. This can be used to debug high level C# code.
+More details [here](
+https://github.com/deislabs/mystikos/tree/main/tests/dotnet-sos#readme
+).
+
+### Known limitations
+- GC suspension signals are not delivered in a real-time fashion. Actively being addressed.
+
+### Library Support
+TODO
```
Comment (jxyang) [inline doc/dotnet-support.md:36]: I think we should provide a sample docker file and a sample config.json to put the recommendations together.

Comment (vtikoo) [reply doc/dotnet-support.md:36]: We have a sample docker file in the [getting started dotnet document](https://github.com/deislabs/mystikos/blob/main/doc/user-getting-started-docker-dotnet.md#containerize-the-program). Its however only for 3.1.
I can add the rest there and link it up here.

File: doc/dotnet-support.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+- Alpine
+- Ubuntu
+
+We recommend using official images from [mcr.microsoft.com/dotnet](https://hub.docker.com/_/microsoft-dotnet/).
+
+### Recommended configurations
+
+#### dotnet configurations
+- Run with dotnet diagnostics disabled. This can be controlled via the `COMPlus_EnableDiagnostics` environment variable set to 0.
+- If running with low memory settings, you can control dotnet's managed heap size by [`COMPlus_GCHeapHardLimit` environment variable](https://docs.microsoft.com/en-us/dotnet/core/run-time-config/garbage-collector#heap-limit).
```
Comment (jxyang) [inline doc/dotnet-support.md:16]: As discussed in the meeting, we should also experiment with  building the dotnet app with or without ``self-contained` flag and see the impacts on the memory requirement.

**Review by anakrish — COMMENTED** [review]
It is important to mention that `dotnet test` does not work well due to the use of two levels for vforking.

File: doc/dotnet-support.md (hunk 3 above)
Comment (jxyang) [reply doc/dotnet-support.md:16]: Ignore the previous comment. My experiment shows `self-contained` is not relevant to memory requirement.

Comment (vtikoo) [reply doc/dotnet-support.md:16]: Resolving based on your Teams message that `self-contained` doesn't impact memory requirement.

**Comment by vtikoo** [comment]
> It is important to mention that `dotnet test` does not work well due to the use of two levels for vforking.

Do you have any specific issues in mind? Given that your CRT fix should hopefully address the memory corruption failures.
I can think of only larger memory requirement(~6G) in a two-level fork.

File: doc/dotnet-support.md (hunk 1 above)
Comment (vtikoo) [reply doc/dotnet-support.md]: Added

**Comment by anakrish** [comment]
There are few issues I can think of with `dotnet test`
- dotnet-sos doesn't work 
- Currently `dotnet test` is broken since dup is not supported for AF_LOCAL sockets
- We don't lock `dotnet test` anymore in CI. Therefore it is not tested.

**Review by jxyang — APPROVED** [review]

**Comment by vtikoo** [comment]
> * dotnet-sos doesn't work

Added a line about this.

> * Currently `dotnet test` is broken since dup is not supported for AF_LOCAL sockets

Is this going to be addressed soon? If its a fundamental limitation going forward, we should definitely document it in more detail.

**Pull request merged**